sdk/aws-autoloader.php

```diff
@@ -2826,8 +2826,13 @@
     }
 }, true);
 
-require __DIR__ . '/Aws/functions.php';
-require __DIR__ . '/GuzzleHttp/functions_include.php';
+if ((!isset($CFG->libraries) && !file_exists($CFG->libraries . '/optional/aws/aws-sdk-php/src/functions.php'))
```

Comment:
```suggestion
if ((!isset($CFG->libraries) || !file_exists($CFG->libraries . '/optional/aws/aws-sdk-php/src/functions.php'))
```
The `AND` should probably be an `OR`, otherwise the two statements are mutually inclusive.

Reply:
Been trying to do this in-between other work but I won't be able to review this week as I'm doing upgrade prep for the rest of the week then on leave next week, so could someone else pick it up perhaps? Happy to give access to my repo for any volunteers.